File: Backend/mapping/src/main/java/com/mhp/coding/challenges/mapping/models/db/blocks/ImageBlock.java
package com.mhp.coding.challenges.mapping.models.db.blocks;

import com.mhp.coding.challenges.mapping.models.db.Image;
import com.mhp.coding.challenges.mapping.models.dto.ImageDto;

public class ImageBlock extends ArticleBlock {

    private Image image;

    public Image getImage() {
        return image;
    }

    public void setImage(Image image) {
        this.image = image;
    }
    
    @Override
    public com.mhp.coding.challenges.mapping.models.dto.blocks.ImageBlock map() {

        com.mhp.coding.challenges.mapping.models.dto.blocks.ImageBlock blockDto
          = new com.mhp.coding.challenges.mapping.models.dto.blocks.ImageBlock();

        blockDto.setSortIndex(this.getSortIndex());
        
        if (this.getImage() != null)
            blockDto.setImage(this.getImage().map());
        else
            blockDto.setImage(new ImageDto());
        
        return blockDto;
    }
}

File: Backend/mapping-kotlin/src/main/kotlin/com/mhp/coding/challenges/mapping/models/db/blocks/ImageBlock.kt
package com.mhp.coding.challenges.mapping.models.db.blocks

import com.mhp.coding.challenges.mapping.models.db.Image
import com.mhp.coding.challenges.mapping.models.db.ImageSize
import com.mhp.coding.challenges.mapping.models.dto.ImageDto

class ImageBlock(
    var image: Image?,
    override val sortIndex: Int = 0,
) : ArticleBlock(sortIndex) {

    override fun map():
        com.mhp.coding.challenges.mapping.models.dto.blocks.ImageBlock {
    
        val _image : Image? = this.image
        val imageBlockDto =
            com.mhp.coding.challenges.mapping.models.dto.blocks.ImageBlock(
            
            image = if (_image != null) _image.map() else ImageDto(0, "", ImageSize.SMALL),
            sortIndex = this.sortIndex
        )
        
        return imageBlockDto
    }
}
